### DISCUSSION TITLE
how to call a method inside a livewire form object using $wire

### DISCUSSION THREAD

**Discussion by ahmedsayedabdelsalam** [livewire/livewire#8177] [1. Help] [answered]
I have a livewire component that contains a form object `postForm` when I call a method inside it using $wire.postForm.methodName() it throws an error that the method is not found

**Comment by danie-ramdhani** [comment, marked as answer]
try do it in the main component:

```php
public function doSomething()
{
    $this->postForm->methodName();
}
```

```blade
<button wire:click="doSomething">
    Do Something
</button>
```

**Reply by joshhanley** [reply]
@ahmedsayedabdelsalam yeah @danie-ramdhani is correct, you can't call methods of properties from the front end and that could potentially lead to security issues. Just make a method in your component and use it to call the method on the form.

**Reply by ahmedsayedabdelsalam** [reply]
thanks @danie-ramdhani yeah i know that will work but i thought the form object was for separating the form logic there, if need to duplicate it inside the main component why should i use the form object then?
